### PR TITLE
Jack/endpoints

### PR DESCRIPTION
**Pull request by jackthmp**
Reorganized endpoint requests into files instead of folders. #417

### CODE REVIEW

File: Podcast/ListeningHistoryEndpointRequests.swift [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+import Foundation
+import SwiftyJSON
+
+class FetchListeningHistoryEndpointRequest: EndpointRequest {
+    
+    var offset: Int
+    var max: Int
+    
+    //dismissed is an optional argument that if supplied will filter the
```
Comment (myo3) [inline Podcast/ListeningHistoryEndpointRequests.swift]: add space after `//`

File: Podcast/FacebookEndpointRequests.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+    
+    override func processResponseJSON(_ json: JSON) {
+        processedResponseValue = json["data"]["users"].map{ element in User(json: element.1) }
+    }
+}
+
+class DismissFacebookFriendEndpointRequest: EndpointRequest {
+    
+    var facebookId: String // id of the friend we are dismissing
```
Comment (myo3) [inline Podcast/FacebookEndpointRequests.swift]: put comment above `var facebookId` for consistency with other comments

File: Podcast/UserSubscriptionsEndpointRequests.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        super.init()
+        
+        path = "/subscriptions/users/\(userID)/"
+        
+        httpMethod = .get
+    }
+    
+    override func processResponseJSON(_ json: JSON) {
+        let series = json["data"]["subscriptions"].map{ jsons -> Series in
+            let s = Cache.sharedInstance.update(seriesJson: jsons.1["series"])
```
Comment (myo3) [inline Podcast/UserSubscriptionsEndpointRequests.swift]: more descriptive variable name than `s`

**Review by myo3 — COMMENTED** [review]
Nice clarifying comments 😊 . Only comment is line breaks inconsistent across the files. E.g. `CreateBookmarkEndpointRequest` has no line breaks in the `init` but `FetchUserSubscriptionsEndpointRequest` has line break after every line. Another ex: `AuthenticateUserEndpointRequest` doesn't start w line break in `init` but `processResponseJSON` starts w line break. Once you standardize the line breaks across all files, I think it's good to go

**Review by myo3 — COMMENTED** [review]
Nice clarifying comments 😊 . Only comment is line breaks inconsistent across the files. E.g. `CreateBookmarkEndpointRequest` has no line breaks in the `init` but `FetchUserSubscriptionsEndpointRequest` has line break after every line. Another ex: `AuthenticateUserEndpointRequest` doesn't start w line break in `init` but `processResponseJSON` starts w line break. Once you standardize the line breaks across all files, I think it's good to go

File: Podcast/FetchByIDEndpointRequests.swift [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    init(seriesID: String, offset: Int, max: Int) {
+        
+        self.seriesID = seriesID
+        self.offset = offset
+        self.max = max
+        
+        super.init()
+        
+        path = "/podcasts/episodes/by_series/\(seriesID)/"
+
```
Comment (AAAstorga) [inline Podcast/FetchByIDEndpointRequests.swift]: could group these 3 together without line breaks in between. Just a suggestion

File: Podcast/RecommendationsEndpointRequests.swift [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        }
+        processedResponseValue = episodes
+    }
+}
+
+class CreateRecommendationEndpointRequest: EndpointRequest {
+    
+    var episodeID: String
+    
+    init(episodeID: String, with blurb: String? = nil) {
```
Comment (AAAstorga) [inline Podcast/RecommendationsEndpointRequests.swift:38]: is blurb the right choice here? Can be a little confusing on what the blurb is

**Review by myo3 — COMMENTED** [review]
Nice clarifying comments 😊 . Only comment is line breaks inconsistent across the files. E.g. `CreateBookmarkEndpointRequest` has no line breaks in the `init` but `FetchUserSubscriptionsEndpointRequest` has line break after every line. Another ex: `AuthenticateUserEndpointRequest` doesn't start w line break in `init` but `processResponseJSON` starts w line break. Once you standardize the line breaks across all files, I think it's good to go

File: Podcast/SearchEndpointRequests.swift [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    override func processResponseJSON(_ json: JSON) {
+        let episodes = json["data"]["episodes"].map{ episodeJSON in
+            Cache.sharedInstance.update(episodeJson: episodeJSON.1)
+        }
+        let series = json["data"]["series"].map{ seriesJSON in
+            Cache.sharedInstance.update(seriesJson: seriesJSON.1)
+        }
+        let users = json["data"]["users"].map{ userJSON in
+            Cache.sharedInstance.update(userJson: userJSON.1)
+            }.filter{
```
Comment (AAAstorga) [inline Podcast/SearchEndpointRequests.swift]: space after filter and before the bracket `.filter {`

File: Podcast/UserAuthEndpointRequests.swift [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        
+        let sessionJSON = json["data"]["session"]
+        let session = Session(json: sessionJSON)
+        
+        processedResponseValue = ["user": user, "session": session]
+    }
+}
+
+enum SignInType {
+    case Facebook
```
Comment (AAAstorga) [inline Podcast/UserAuthEndpointRequests.swift]: enums should be camel case

File: Podcast/UserSubscriptionsEndpointRequests.swift (hunk 3 above)
Comment (AAAstorga) [reply Podcast/UserSubscriptionsEndpointRequests.swift]: can't you just do ` return Cache.sharedInstance.update...`

**Review by AAAstorga — DISMISSED** [review]
Awesome stuff! Looks very good, just had some comments. Really minor stuff. Once you fix you can merge 💯

File: Podcast/DiscoverEndpointRequests.swift [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+            processedResponseValue = json["data"][requestType.rawValue].map{ seriesJSON in
+                Cache.sharedInstance.update(seriesJson: seriesJSON.1)
+            }
+        }
+    }
+    
+}
+
+class GetAllTopicsEndpointRequest: EndpointRequest {
+
```
Comment (doctor3w) [inline Podcast/DiscoverEndpointRequests.swift]: Could you standardize these line breaks? They are all different from over the years.

File: Podcast/FacebookEndpointRequests.swift [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        self.max = max
+        
+        super.init()
+        
+        path = "/search/facebook/friends/\(query.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed) ?? query)/"
+        httpMethod = .get
+        queryParameters = ["offset": offset, "max": max]
+        headers = ["AccessToken": facebookAccessToken]
+        
+
```
Comment (doctor3w) [inline Podcast/FacebookEndpointRequests.swift]: Like this space here

File: Podcast/FetchByIDEndpointRequests.swift (hunk 4 above)
Comment (doctor3w) [reply Podcast/FetchByIDEndpointRequests.swift]: plus one this suggestion

File: Podcast/FollowsEndpointRequests.swift [hunk 10]
```diff
@@ -52,5 +52,43 @@ class FetchUserFollowsByIDRequest: EndpointRequest {
...
+    
+}
+
+class FollowUserEndpointRequest: EndpointRequest {
+    
+    // ID to follow
+    var userID: String
+    
+    init(userID: String) {
+
```
Comment (doctor3w) [inline Podcast/FollowsEndpointRequests.swift]: Line breaks. Sorry! We were not consistent the first time around.

File: Podcast/RecommendationsEndpointRequests.swift (hunk 5 above)
Comment (doctor3w) [reply Podcast/RecommendationsEndpointRequests.swift:38]: Blurb is our app verbiage. A recast with a blurb is our way of "Retweeting a podcast with a comment"

File: Podcast/SearchEndpointRequests.swift (hunk 6 above)
Comment (doctor3w) [reply Podcast/SearchEndpointRequests.swift]: Should we do this for all the maps above as well?

File: Podcast/ShareEndpointRequests.swift [hunk 11]
```diff
@@ -1,43 +1,43 @@
...
     var offset: Int
     var max: Int
-
+    
     init(offset: Int, max: Int) {
         self.offset = offset
         self.max = max
         super.init()
-
+
```
Comment (doctor3w) [inline Podcast/ShareEndpointRequests.swift:21]: Please delete this extra whitespace, it occurs a lot in this file

File: Podcast/UserSubscriptionsEndpointRequests.swift (hunk 3 above)
Comment (doctor3w) [reply Podcast/UserSubscriptionsEndpointRequests.swift]: Yeah I don't think you even return. This code used to do some weird casting stuff but now it's unnecessary

File: Podfile.lock [hunk 12]
```diff
@@ -135,4 +108,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 44fbb8afb3d39fedd83bc6bfb53b350b452d21c8
 
-COCOAPODS: 1.5.0
+COCOAPODS: 1.4.0
```
Comment (doctor3w) [inline Podfile.lock]: please update your cocoapods installation. Podfile.lock SHOULD NOT be changed.

**Review by doctor3w — CHANGES_REQUESTED** [review]

File: Podcast/SearchEndpointRequests.swift (hunk 6 above)
Comment (AAAstorga) [reply Podcast/SearchEndpointRequests.swift]: Yeah, I believe swiftlint might automatically fix this

File: Podcast/ShareEndpointRequests.swift (hunk 11 above)
Comment (AAAstorga) [reply Podcast/ShareEndpointRequests.swift:21]: Swiftlint will automatically remove white space. You guys should install it!

File: Podcast/FacebookEndpointRequests.swift [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+        self.pageSize = pageSize
+        self.returnFollowing = returnFollowing
+        super.init()
+        
+        path = "/users/facebook/friends/"
+        httpMethod = .get
+        queryParameters = ["offset": offset, "max": pageSize]
+        
+        if let following = returnFollowing {
+            queryParameters!["return_following"] = encodeBoolean(following)
```
Comment (AAAstorga) [inline Podcast/FacebookEndpointRequests.swift]: what is this force cast. Make it safe plz

File: Podcast/ListeningHistoryEndpointRequests.swift [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+        self.offset = offset
+        self.max = max
+        self.dismissed = dismissed
+        super.init()
+        
+        path = "/history/listening/"
+        httpMethod = .get
+        queryParameters = ["offset": offset, "max": max]
+        if let dismiss = dismissed {
+            queryParameters!["dismissed"] = encodeBoolean(dismiss)
```
Comment (AAAstorga) [inline Podcast/ListeningHistoryEndpointRequests.swift]: same here

Comment (AAAstorga) [reply Podcast/ListeningHistoryEndpointRequests.swift]: potentially make encodeBoolean an extension of Bool so you can do `dismiss.encode()`

**Review by AAAstorga — DISMISSED** [review]
3 small comments, otherwise looks good

**Review by doctor3w — DISMISSED** [review]

**Review by doctor3w — APPROVED** [review]

**Pull request merged**